player.py

```diff
@@ -77,6 +77,14 @@ async def skip(self, ctx):
         else:
             await ctx.respond("There is no queue my friend")
 
+    #code to add a stop music and disconnect command
+    async def stop_and_disconnect(self):
+        if self._voice_client.is_playing():
+            self._voice_client.stop()
+        if self._voice_client.is_connected():
+            await self._voice_client.disconnect()
+        self._queue.clear()
```

Comment:
You should have only line 85 in here. If this object exists, it's connected to a voice chat. You can leave line 84 if you want, but it shouldn't be necessary. After the disconnect this object is deleted automatically, therefore you can remove line 86.